src/main/js/SaveApi.js

```diff
@@ -6,22 +6,23 @@ import { Fetch, getRestUrl, sseService, loadingIndicator, capabilityAugmenter }
 export class SaveApi {
 
     indexRepo(organization, teamName, repoName) {
-        const createUrl = `${getRestUrl({organization})}/pipelines/`;
-
+        const createUrl = repoName ? `${getRestUrl({organization})}pipelines/` : `${getRestUrl({organization, pipeline: teamName})}`;
```

Comment:
No it won't because scmId in that case will be 'git'. scmId on [line 407](https://github.com/jenkinsci/blueocean-pipeline-editor-plugin/blob/master/src/main/js/EditorPage.jsx#L407) is GitHub or GitHub enterprise only in case  when GitHub or GitHub enterprise based jobs are org folder.